File: CodeVita_2020/Railway_Station.cpp
#include<bits/stdc++.h>
using namespace std;

/*int main()
{
    int n;
    cin>>n;
    int time[86400*2 + 1]={0};
    int ans=0,maxtime=0;
    //multimap <int,int> m;
    for(int i=0;i<n;i++)
    {
        int n1,n2;
        cin>>n1>>n2;
        //m.insert(pair <int, int> (n1, n2));
        for(int i=n1;i<=(n1+n2);i++)
        {
            time[i]++;
            ans = ans>time[i]?ans:time[i];
        }
    }
    cout<<ans;
    //int plat = 0,ans = 0,trains=0;
    return 0;
}*/

int main()
{
    int n;
    cin>>n;
    int arr[n];
    int dept[n];
    for(int i=0;i<n;i++)
    {
        int n1,n2;
        cin>>n1>>n2;
        arr[i] = n1;
        dept[i] = n1+n2;
    }
    sort(arr,arr+n);
    sort(dept,dept+n);
    int i=1,plat=1,j=0,maxplat=1;
     while(i<n && j<n)
     {
         if(arr[i]<=dept[j])
         {
             plat++;
             if(plat>maxplat)
             {
                 maxplat = plat;
             }
             i++;
         }
         else
         {
             plat--;
             j++;
         }
     }
     cout<<maxplat;
    return 0;
}

File: CodeVita_2020/railwaystation.py
if __name__ == '__main__':
	n = int(input())
	arr = list()
	depa = list()
	for i in range(0,n):
		x =input()
		#print(x)
		num = x.split(" ")
		#print(num)
		n1 = int(num[0])
		n2 = int(num[1])
		arr.append(n1)
		depa.append((n2+n1))
	maxi = 0
	plat = 1
	arr.sort()
	depa.sort()
	i,j = 1,0
	while i < n and j < n:
		if arr[i] <= depa[j]:
			i = i+1
			plat = plat +1
			if maxi < plat: maxi = plat
		else:
			j = j+1
			plat = plat -1
	print(maxi,end=" ")
